src/query/service/src/spillers/spiller.rs

```diff
@@ -114,31 +122,17 @@ impl Spiller {
     }
 
     /// Spill a [`DataBlock`] to storage.
-    pub async fn spill(&mut self, data_block: DataBlock) -> Result<String> {
-        // Serialize data block.
-        let (data_size, columns_data, columns_layout) = self.serialize_data_block(data_block)?;
+    pub async fn spill(&mut self, data_block: DataBlock) -> Result<Location> {
+        let instant = Instant::now();
 
         // Spill data to storage.
-        let instant = Instant::now();
-        let unique_name = GlobalUniqName::unique();
-        let location = format!("{}/{}", self.config.location_prefix, unique_name);
-        let mut writer = self
-            .operator
-            .writer_with(&location)
-            .chunk(8 * 1024 * 1024)
-            .await?;
-        for data in columns_data.into_iter() {
-            writer.write(data).await?;
-        }
-        writer.close().await?;
+        let encoded = EncodedBlock::from_block(&data_block);
+        let columns_layout = encoded.columns_layout();
+        let data_size = encoded.size();
+        let location = self.write_encodes(data_size, vec![encoded]).await?;
 
         // Record statistics.
-        Profile::record_usize_profile(ProfileStatisticsName::SpillWriteCount, 1);
-        Profile::record_usize_profile(ProfileStatisticsName::SpillWriteBytes, data_size as usize);
-        Profile::record_usize_profile(
-            ProfileStatisticsName::SpillWriteTime,
-            instant.elapsed().as_millis() as usize,
-        );
+        record_write_profile(&instant, data_size);
```

Comment:
It would be better to record the profile of `LocalSpill` and `RemoteSpill` separately.